Week-3/q1/public/js/index.js

```diff
@@ -0,0 +1,40 @@
+//Given reducer method:
+function add(a, b) {
+	return a + b;
+}
+
+// Create a memoize function
+const memoize = (fn) => {
+	const cache = new Map();
+	return function (...args) {
+		let key = args.toString();
+
+		/**
+		 * Check if we have already passed the same arguments
+		 * If yes, return value
+		 */
+		if (cache.has(key)) {
+			return cache.get(key);
+		}
+
+		/**
+		 * User reduce method to get total
+		 */
+		let total = args.reduce((acc, current) => fn(acc, current), 0);
```

Comment:
No need to use reduce here, just call the fn(...args) that is enough

Reply:
Actually, we are getting an array, That's why I have used reduce for iteration, or we could use for loop.